### PR TITLE
allow users to get their own status

### PR DESCRIPTION
**Pull request by MatthewBemis**
This takes care of [DSDEEPB-2104](https://broadinstitute.atlassian.net/browse/DSDEEPB-2104) and [DSDEEPB-2105](https://broadinstitute.atlassian.net/browse/DSDEEPB-2105) as an effort to unblock [DSDEEPDB-2046](https://broadinstitute.atlassian.net/browse/DSDEEPB-2046) for Silver team.
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [ ] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/resources/swagger/rawls.yaml [hunk 1]
```diff
@@ -12,145 +12,6 @@ info:
...
-        - in: path
-          description: userId to delete
-          name: userId
-          required: true
-          type: string
-      security:
-        - authorization:
-            - openid
-            - email
-            - profile
```
Comment (MatthewBemis) [inline src/main/resources/swagger/rawls.yaml:46]: This is old swagger that I forgot to remove. It's unrelated to this PR but I figured I'd remove it before I forgot

File: src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 2]
```diff
@@ -33,7 +33,8 @@ object UserService {
   case object GetRefreshTokenDate extends UserServiceMessage
 
   case object CreateUser extends UserServiceMessage
-  case class GetUserStatus(userRef: RawlsUserRef) extends UserServiceMessage
+  case class AdminGetUserStatus(userRef: RawlsUserRef) extends UserServiceMessage
+  case class UserGetUserStatus() extends UserServiceMessage
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala]: `case object UserGetUserStatus`

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/UserApiService.scala [hunk 3]
```diff
@@ -59,9 +59,16 @@ trait UserApiService extends HttpService with PerRequestCreator with UserInfoDir
       get {
         requestContext => perRequest(requestContext,
           UserService.props(userServiceConstructor, userInfo),
-          UserService.GetUserStatus(RawlsUserRef(RawlsUserSubjectId(userSubjectId))))
+          UserService.AdminGetUserStatus(RawlsUserRef(RawlsUserSubjectId(userSubjectId))))
       }
     } ~
+      path("user") {
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/UserApiService.scala]: shift 1 tab to the left

**Comment by jmthibault79** [comment]
:+1: with small changes

**Comment by MatthewBemis** [comment]
PR comments addressed

**Comment by jmthibault79** [comment]
Groovy.

**Comment by dvoet** [comment]
:+1:

**Pull request merged**